### ISSUE TITLE
New "any height" images doesn't play nicely with portrait images

### ISSUE THREAD

**Issue by IBBoard** [baedert/corebird#465]
Commit 97c626dcefe6683683003046dbc832d757ba3b8f solves a problem of making excessively tall and narrow images not appear too small, skinny and unreadable, but instead it makes them cause lots of scrolling in the feed.

![screenshot from 2015-12-14 09-34-12](https://cloud.githubusercontent.com/assets/388582/11777389/3c8098d0-a246-11e5-8dc0-23c642b96cda.png)

(The original one I saw this with was a "wilw" RT of https://twitter.com/bonniegrrl/status/676130518308544512)

Given that we've got the "click and see full picture" (as Twitter's web interface has) Would it be better to keep the max height and crop the middle for images that are too big? It'd allow sensible dimensions on the preview so that images don't make feeds too long, and landscape images would hopefully still show up in full (depending on their proportions and how wide someone made the window).

**Comment by baedert** [comment]
We can bikeshed for years about image rendering. There's no perfect way to do it.

**Comment by IBBoard** [comment]
True, but as the commit says "let's see how this works out" then I thought I'd put in a ticket about my thoughts and how it works for me.

I was thinking that a 4:3 ratio max limit might work, but then some images aren't that conventional, so then I wondered whether a square maximum would work - that way those with a narrower window don't get huge images and those who want to fill all their screen estate get big images.

I'll try to give it a prod at some point and see if I can get a working approach as an example.

**Comment by Flat** [comment]
Any progress on this issue? I really like corebird but being able to only see one tweet at a time if there's an image is not optimal. As seen here 
![2016-08-08-19 00 55](https://cloud.githubusercontent.com/assets/2048861/17499293/7de178f2-5d9a-11e6-90cd-ff606d46dec3.png)

I see there's an open pull request #512, Is there anything that needs to be changed/added or has it just not been reviewed yet?

**Comment by baedert** [comment]
I don't see the big problem with seeing only one tweet, really. The current image handling is a result of me trying to not do that, but people kept complaining about not being able to see the entire image.

**Comment by IBBoard** [comment]
I think the original problem was that the image was a really thin strip, which showed very little. Then (IIRC) the image preview grew larger, but it still cropped strangely (something about the max height and the fact it didn't scale beyond a set limit?). Now it just shows the full thing.

My patch (which is in my custom build on the openSUSE Build Service - available to anyone with openSUSE 42.1 or Tumbleweed) does 4:3 scaling, so the wider you make the window then the bigger the images can become. Use a narrow window? Images won't take up much space. Like big images? Widen your window. 4:3 seemed like a reasonable compromise.

The only thing I keep thinking I need to add is some kind of styling indicator to show that an image has been cropped, which might help with the "I can't see the entire image" complaints (which are slightly daft, since Twitter already does that anyway, and people like the Royal Institution use it as a *feature* for [doing quizzes with the answers as images](https://twitter.com/Ri_Science/status/748900914883399680))

**Comment by IBBoard** [comment]
I think the general approach is to limit the number of config settings (which is generally sane - too many apps have too many options).

I don't have a second monitor, but I prefer the cropped images for the same reason. There are lots of images that can be posted that I don't care about, so being sensibly cropped doesn't lose me anything, but if there has been a flood of tweets then I don't want to spend too long *just* scrolling past the images (but I still want to see the image, so that I can see whether to view it large or not!). My tweak crops and centres, although it keeps the Corebird behaviour of having multiple images in a single row, rather than Twitter's "one large image with three down the side" layout.

As for not knowing Vala: I didn't know it when I started! But if you know programming (particularly something like Java or C#) then it is easy to pick up. The hardest part appears to be the GTK API (so some prior GTK/C experience probably helps - of which I had basically none!)

**Comment by IBBoard** [comment]
I've not done a pull request for it yet, but I've got a first cut "what could we do to show images are cropped" branch. All it does is put a gradient over the top and bottom, as if it is shaded/disappearing out of a box.

Branch at https://github.com/IBBoard/corebird/tree/bug465-cropped-image-indicator-1 for anyone who wants to try it/use it.

![Cropped and non-cropped in main timeline](https://cloud.githubusercontent.com/assets/388582/17601993/9592a89e-6002-11e6-86f3-c605f2583a66.png) 
![Cropped image in single tweet](https://cloud.githubusercontent.com/assets/388582/17602015/ad8d8cd4-6002-11e6-9d35-09c42a19a3bc.png)

**Comment by siebenmann** [comment]
As an opinion point: I tried out a version of corebird with @IBBoard 's change and found that I very much didn't like it as much as the current behavior of showing the entire image. With the cropped images, I found that I kept wanting to click them to see if there was anything important I was missing in the cut-off areas. With un-cropped images in regular corebird I could simply scan through my feed and decide confidently and immediately that no, this image (or video or etc) wasn't interesting enough to click on for a better version of it.

(As a result, my bias is that if a more compact display is desired it should be done by scaling down the entire image to a larger degree than is currently done. But my overall preference is the current behavior as it is.)

**Comment by IBBoard** [comment]
Oddly, I don't seem to notice all that many cropped images, and I can normally tell enough to skip a lot of them anyway.

Mine and Flat's examples are why we think there needs to be some kind of cropping. I might look at it again and see how bad height-capping and scaling works instead.

The one advantage of my cropping is that it matches Twitter's behaviour (albeit with an actual indication that something was hidden!). While that's not the best reason for cropping, it does mean that anyone making a tweet that makes use of cropping loses out (e.g. [this thread](https://twitter.com/IBBoard/status/614483807236919296)). That said, that was a while ago and I'm not sure that I've seen people use it that way recently.

**Comment by IBBoard** [comment]
I've adjusted my patch so that it now crops at 9:16 rather than 4:3. That means that portrait smartphone pics should still show uncropped, and I can still see more than one picture with the width that I have Corebird set to.

https://build.opensuse.org/package/view_file/home:IBBoard:desktop/corebird/bug465-portrait-image-sizes.patch?expand=1

Is that any better? It's less intrusive now and more of a "catch the worst case" thing.
